### PR TITLE
Add .MonthLastDay() to schedule a task at the end of each month

### LINKED ISSUE #224: [FEATURE] - Monthly schedules for ending of month
### Describe the solution you'd like

We should be able to schedule jobs that will run at the end of each month, as in "last day of each month", "penultimate day of the month", or "antepenultimate day of month", which covers all cases for feb 28th to 31 days months.

Ideas just to get things started:

- `.Months(gocron.MonthlyLastDay)`, (or `.MonthLastDay()`)
- `.Months(gocron.MonthlyPenultimateDay)` (or `.Months(gocron.SecondLastDay)`, `.MonthSecondLastDay()`?)
- `.Months(gocron.MonthlyAntepenultimateDay)` (or `.Months(gocron.ThirdLastDay)`, `.MonthThirdLastDay()`?)

related #141, #221

### PR DESCRIPTION
**Pull request by mrngm**
### What does this do?

This PR adds the `.MonthLastDay()` function to indicate a job needs to run every last day of the month. It introduces the special value `-1` for `.Months()` to indicate the last day of the month. It is not allowed to use `-1` in a list of days, e.g. `.Months(12, 15, -1)` is invalid.

### Which issue(s) does this PR fix/relate to?

Resolves #224

### List any changes that modify/break current functionality

* `Scheduler.Months(-1)` is now a valid call

### Have you included tests for your changes?

Yes, see changes on `scheduler_test.go`.

### Did you document any new/modified functionality?

- [x] Updated `example_test.go`
- [x] Updated `README.md`

### Notes

* `make test` appears to be non-deterministic in `TestScheduler_Update` and sometimes fails in `happy_path` or `update,_delete,_create`, or it simply passes. As far as I can tell, this PR did no changes in the code paths for these tests.
* `make lint` (although `golint` has been [deprecated](https://github.com/golang/lint#readme)) outputs errors unrelated to this PR:
```
$ make lint
~/gocron/scheduler.go:1054:1: exported method Scheduler.Cron should have comment or be unexported
~/scm/gocron/scheduler.go:1058:1: exported method Scheduler.CronWithSeconds should have comment or be unexported
Found 2 lint suggestions; failing.
make: *** [Makefile:27: lint] Error 1
```
* Instead of `-1`, some (large) constant could be introduced instead. Please let me know if that is preferable.
* `make vet` passes
* `make fmt` has been run

### CODE REVIEW

**Comment by JohnRoesler** [comment]
@mrngm thank you for the contribution!

- Appreciate the call outs on the Makefile. Looks like that needs some updates. I opened an issue for that https://github.com/go-co-op/gocron/issues/251
- -1 is good for the end of month value

File: example_test.go [hunk 1]
```diff
@@ -397,6 +397,13 @@ func ExampleScheduler_Monday() {
 	// Monday
 }
 
+func ExampleScheduler_MonthLastDay() {
```
Comment (JohnRoesler) [inline example_test.go:400]: the alphabetize check is complaining that this should go underneath the Month example on 407

File: scheduler_test.go [hunk 2]
```diff
@@ -751,6 +754,15 @@ func TestScheduler_CalculateNextRun(t *testing.T) {
...
+		{name: "every last day of the month started on non-leap year february should run on march 31", job: &Job{interval: 1, unit: months, daysOfTheMonth: []int{-1}, lastRun: time.Date(2019, time.February, 28, 0, 0, 0, 0, time.UTC)}, wantTimeUntilNextRun: 31 * day},
+		{name: "every last day of 2 months started on leap year february should run on april 30", job: &Job{interval: 2, unit: months, daysOfTheMonth: []int{-1}, lastRun: time.Date(2020, time.February, 29, 0, 0, 0, 0, time.UTC)}, wantTimeUntilNextRun: 31*day + 30*day},
+		{name: "every last day of 2 months started on non-leap year february should run on april 30", job: &Job{interval: 2, unit: months, daysOfTheMonth: []int{-1}, lastRun: time.Date(2019, time.February, 28, 0, 0, 0, 0, time.UTC)}, wantTimeUntilNextRun: 31*day + 30*day},
+		{name: "every last day of the month started on january 1 in leap year should run on february 29", job: &Job{interval: 1, unit: months, daysOfTheMonth: []int{-1}, lastRun: januaryFirst2020At(0, 0, 0)}, wantTimeUntilNextRun: 30*day + 29*day},
```
[the remark this answers is not in the document]
Comment (mrngm) [reply scheduler_test.go:761]: Good call. I've added test cases for this as well.

**Comment by mrngm** [comment]
@JohnRoesler I didn't notice you updated my fork's branch; I rebased my commits on top of your merge commit. I hope that's alright.

Let me know if there's anything else that needs to be done for this PR.

**Comment by Streppel** [comment]
@mrngm thanks for your contribution! As of end of months, we actually might have 3 edge cases, see if you agree with me: each of the last 3 days of the month. If so, wouldn't it be worth to consider it in this method as well? I was thinking maybe going for an integer as a parameter as in MonthLastDay(1), MonthLastDay(2) etc. wdyt?

**Comment by mrngm** [comment]
> @mrngm thanks for your contribution! As of end of months, we actually might have 3 edge cases, see if you agree with me: each of the last 3 days of the month. If so, wouldn't it be worth to consider it in this method as well? I was thinking maybe going for an integer as a parameter as in MonthLastDay(1), MonthLastDay(2) etc. wdyt?

In this PR, I've worked around those edge cases, because the last day of the month can actually take 4 values (28-31). I think it's unreasonable to require a user of `gocron` to think about what the last day of the month actually is, taking into account the starting date, and the possibility of a leap year.

By using the first day of the _next_ month (and subtracting one day), we always get the correct last day of the month (that is, the day we want to schedule the job), regardless of starting date or whether we're in a leap year.

For this PR, I would not like to add additional functions to specify "some day before the last day of the month". Do you have specific use cases that would require this functionality?

**Review by JohnRoesler — APPROVED** [review]

**Pull request merged**